src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+
+    public Figure getRandomFigure() {
+        ColorSupplier randomColor = new ColorSupplier();
+        Random randomIndex = new Random();
+
+        switch (randomIndex.nextInt(NUMBER_OF_FIGURES)) {
+            case 0:
+                return new Circle(randomColor.getRandomColor(),
+                        getRandomDouble());

```

Comment:
```suggestion
                return getRandomCircle();
```
lets create separate private methods to get random figures